File: Data Structures & Algorithms/matrixDFS/submission-0.java
class Solution {
    public int countPaths(int[][] grid) {

        //int ROW = grid.length;
        //int COL = grid[0].length;

        Set<String> set = new HashSet<String>();
        return helper(grid, 0, 0, set);

    }


    public int helper(int[][] grid, int r, int c, Set<String> set) {

        if(Math.min(r,c) < 0 || r == grid.length || c == grid[0].length) 
            return 0;

        if (set.contains(r + "." + c) || grid[r][c] == 1)
            return 0;
        
        // when does it return 1
        if(r == grid.length - 1 && c == grid[0].length - 1)
            return 1;

        set.add(r + "." + c);
        int count = 0;
        count = count + helper(grid, r+1, c, set);
        count = count + helper(grid, r, c+1, set);
        count = count + helper(grid, r-1, c, set);
        count = count + helper(grid, r, c-1, set);

        set.remove(r + "." + c);

        return count;
    }
}

File: Data Structures & Algorithms/count-servers-that-communicate/submission-0.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        ROW = len(grid)
        COL = len(grid[0])

        row_count = [0]*ROW
        col_count = [0]*COL

        for r in range(ROW):
            for c in range(COL):
                if grid[r][c] == 1:
                    row_count[r] += 1
                    col_count[c] += 1

        res = 0

        for r in range(ROW):
            for c in range(COL):
                if grid[r][c] == 1:
                    if row_count[r] > 1 or col_count[c] > 1:
                        res += 1

        return res